File: ecommerce-api-legacy/src/middlewares/errorHandler.js
function errorHandler(err, req, res, next) {
  const status = err.statusCode || 500;
  if (status >= 500) console.error('[ERROR]', err.message, err.stack);
  res.status(status).json({ error: status >= 500 ? 'Erro interno do servidor' : err.message });
}

module.exports = errorHandler;

File: task-manager-api/middlewares/error_handler.py
import logging
from flask import jsonify

logger = logging.getLogger(__name__)


class AppError(Exception):
    status_code = 500


class ValidationError(AppError):
    status_code = 400


class NotFoundError(AppError):
    status_code = 404


class UnauthorizedError(AppError):
    status_code = 401


class ForbiddenError(AppError):
    status_code = 403


class ConflictError(AppError):
    status_code = 409


def register_error_handlers(app):
    @app.errorhandler(AppError)
    def handle_app_error(e):
        return jsonify({'error': str(e)}), e.status_code

    @app.errorhandler(Exception)
    def handle_unexpected(e):
        logger.exception('Erro não tratado')
        return jsonify({'error': 'Erro interno do servidor'}), 500
